File: Project2/store/server/src/models/receiveStock.js
const { PUSHER_CHANNEL_STORE } = require('./../config/configs');
const { messageType } = require('../enums');
const { sendNotificationMessage } = require('../services/websockets/pusher');

module.exports = (sequelize, DataTypes) => {
  const ReceiveStock = sequelize.define('ReceiveStock', {
    quantity: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 0,
    },
    processedDate: {
      type: DataTypes.DATE,
    },
    ordersId: {
      type: DataTypes.JSONB,
    },
  }, {});

  ReceiveStock.associate = (models) => {
    ReceiveStock.belongsTo(models.Book, {
      foreignKey: 'bookId',
    });
  };

  ReceiveStock.afterCreate(async (receiveStock) => {
    const book = await receiveStock.getBook();
    sendNotificationMessage(PUSHER_CHANNEL_STORE, messageType.createReceiveStock, {...receiveStock.dataValues, book});
  });

  ReceiveStock.afterUpdate(async (receiveStock) => {
    const book = await receiveStock.getBook();
    sendNotificationMessage(PUSHER_CHANNEL_STORE, messageType.updateReceiveStock, {...receiveStock.dataValues, book});
  });

  return ReceiveStock;
};

File: Project2/store/interface/GUI/ReceiveStock.cs
namespace @interface
{
    class ReceiveStock
    {
        public int id { get; set; }
        public int quantity { get; set; }
        public string processedDate { get; set; }
        public Book book { get; set; }

        public ReceiveStock()
        {
        }

        public ReceiveStock(dynamic data, Book book)
        {
            id = (int) data.id;
            quantity = (int) data.quantity;
            processedDate = (string) data.processedDate;
            this.book = book;
        }

        public ReceiveStock(int id, int quantity, string processedDate, Book book)
        {
            this.id = id;
            this.quantity = quantity;
            this.processedDate = processedDate;
            this.book = book;
        }
    }
}